src/notificationmanager.cpp

```diff
@@ -53,176 +142,252 @@ NotificationManager::NotificationManager(TDLibWrapper *tdLibWrapper, AppSettings
-
-    bool needFeedback;
-    const AppSettings::NotificationFeedback feedbackStyle = appSettings->notificationFeedback();
-
-    if (activeNotifications.isEmpty()) {
-        QString notificationBody;
-        if (addAuthor) {
-            QVariantMap authorInformation = tdLibWrapper->getUserInformation(messageMap.value("sender_user_id").toString());
-            QString firstName = authorInformation.value("first_name").toString();
-            QString lastName = authorInformation.value("last_name").toString();
-            QString fullName = firstName + " " + lastName;
+    QString notificationBody;
+    if (notificationGroup->totalCount == 1 && !messageMap.isEmpty()) {
+        LOG("Group" << notificationGroup->notificationGroupId << "has 1 notification");
+        if (chatInformation && (chatInformation->type == TDLibWrapper::ChatTypeBasicGroup ||
+           (chatInformation->type == TDLibWrapper::ChatTypeSupergroup && !chatInformation->isChannel))) {
+            // Add author
+            const QVariantMap authorInformation = tdLibWrapper->getUserInformation(messageMap.value(SENDER_USER_ID).toString());
+            const QString firstName = authorInformation.value(FIRST_NAME).toString();
+            const QString lastName = authorInformation.value(LAST_NAME).toString();
+            const QString fullName = firstName + " " + lastName;
             notificationBody = notificationBody + fullName.trimmed() + ": ";
         }
-        notificationBody = notificationBody + this->getNotificationText(messageMap.value("content").toMap());
-        nemoNotification.setBody(notificationBody);
-        needFeedback = (feedbackStyle != AppSettings::NotificationFeedbackNone);
+        notificationBody += getNotificationText(messageMap.value(CONTENT).toMap());
+        nemoNotification->setBody(notificationBody);
     } else {
-        nemoNotification.setReplacesId(activeNotifications.first().toMap().value("replaces_id").toUInt());
-        nemoNotification.setBody(tr("%1 unread messages").arg(activeNotifications.size() + 1));
-        needFeedback = (feedbackStyle == AppSettings::NotificationFeedbackAll);
+        // Either we have more than one notification or we have no content to display
+        LOG("Group" << notificationGroup->notificationGroupId << "has" << notificationGroup->totalCount << "notifications");
+        notificationBody = tr("%1 unread messages").arg(notificationGroup->totalCount);
     }
 
+    nemoNotification->setBody(notificationBody);
+    nemoNotification->setSummary(chatInformation ? chatInformation->title : QString());
     if (needFeedback) {
-        nemoNotification.setCategory("x-nemo.messaging.im");
-        ngfClient->play("chat");
+        nemoNotification->setCategory(NOTIFICATION_CATEGORY);
+        nemoNotification->setPreviewBody(nemoNotification->body());
+        nemoNotification->setPreviewSummary(nemoNotification->summary());
```

Comment:
I probably remove them for now as I explicitly didn't add previewBody/Summary in the original version. They cause notification popups even when I'm in the respective chat. I might add a small 

if (not in chat that receives notification) then (set previewBody/Summary to body() and summary()) else (set previewBody/Summary to QString())

in the future.

Reply:
If you don't set those explicitly, they won't be automatically updated together with body/summary and you get a mismatch.

Reply:
Yep, therefore 86ac99004d170dee622fefc7f7bcebb5bb11d562 sets them now explicitly to an empty string...

Reply:
But yes, if you're already in this chat, a number in a circle in the bottom right corner would be a far more appropriate notification mechanism then a popup.